aws/s3_test.go

```diff
@@ -488,19 +493,34 @@ func bucketNamesForConfigTests() []string {
 func TestFilterS3Bucket_Config(t *testing.T) {
 	t.Parallel()
 
-	// Create test buckets
-	awsParams, err := newS3TestAWSParams()
+	// Create AWS session in ca-central-1
+	awsParams, err := newS3TestAWSParams("ca-central-1")
 	require.NoError(t, err, "Failed to setup AWS params")
 
+	// Nuke all buckets in ca-central-1 first
+	// passing in a config that matches all buckets
+	var configObj *config.Config
+	configObj, err = config.GetConfig("../config/mocks/s3_all.yaml")
+
+	// Verify that only filtered buckets are listed
+	cleanupBuckets, err := getAllS3Buckets(awsParams.awsSession, time.Now().Add(1*time.Hour), []string{awsParams.region}, "", 100, *configObj)
+	require.NoError(t, err, "Failed to list S3 Buckets in ca-central-1")
+
+	nukeAllS3Buckets(awsParams.awsSession, cleanupBuckets[awsParams.region], 1000)
```

Comment:
Is there another way to do this other than 1. listing all the buckets, 2. nuke the listed buckets?

Reply:
AFAIK, this is the model of how cloud nuke works, so unfortunately, no this is the only way for now.